core/src/main/scala/kafka/server/ConfigHelper.scala

```diff
@@ -149,6 +153,19 @@ class ConfigHelper(metadataCache: MetadataCache, config: KafkaConfig, configRepo
                 .setConfigs(configEntries.asJava)
             }
 
+          case ConfigResource.Type.GROUP =>
+            val group = resource.resourceName
+            if (group == null || group.isEmpty) {
+              throw new InvalidRequestException("Group name must not be empty")
+            } else {
+              val groupProps = configRepository.groupConfig(group)
+              val groupConfig = GroupConfig.fromProps(config.groupCoordinatorConfig.extractGroupConfigMap, groupProps)
```

Comment:
I am not sure about this part. My understanding is that it will return the default configs if the group does not have any dynamic configs set. If I remember correctly, it does not work like this for topics. In my mind, it would be better to only return the dynamic configs. It is confusing otherwise. What do you think?

Reply:
Yes, it will return the default configs if the group does not have any dynamic configs set. 
Here I refer to the behavior of the topic. https://github.com/apache/kafka/blob/25b955ddf2253ce76354eee8705155e96ee003fa/core/src/main/scala/kafka/server/ConfigHelper.scala#L106

Reply:
I tried to describe a topic without any dynamic configs with the command line tool and it does not print anything. I wonder if they are filtered somewhere else. It does not work the same for groups. Could you please check?

Reply:
If dynamic configuration is not set, the source of config is `DEFAULT_CONFIG,` and only the configuration with source `DYNAMIC_TOPIC_CONFIG ` will be filtered out in `ConfigCommand`.
https://github.com/apache/kafka/blob/25b955ddf2253ce76354eee8705155e96ee003fa/core/src/main/scala/kafka/admin/ConfigCommand.scala#L593
https://github.com/apache/kafka/blob/25b955ddf2253ce76354eee8705155e96ee003fa/core/src/main/scala/kafka/admin/ConfigCommand.scala#L620

Reply:
![topic-config](https://github.com/user-attachments/assets/188389e6-6b44-4bda-8b7c-4d07707213e7)

Reply:
Thanks. So we always return all the configs when describing. However, we distinguish between default and dynamic entries in the response. How do we differentiate between using DEFAULT_CONFIG or GROUP_CONFIG in our case?

Reply:
I missed this, I have updated this pr.